### PR TITLE
Integrate the 'Buy button' feature

### PR DESCRIPTION
**Pull request by estacioneto**
#### What is the purpose of this pull request?

Add the `buy-button` feature to the product summary

#### What problem is this solving?

A button to reduce the buy workflow, so the user can buy/add to the cart just by clicking in a button

#### How should this be manually tested?
[Access the workspace](https://estacio--storecomponents.myvtex.com/) and see the network tab (dev tools).

#### Screenshots or example usage
image
image

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Changelog
+
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
+and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
+
+## [Unreleased]
```
Comment (brunojdo) [inline CHANGELOG.md]: ⛏ 
### Added 

* **Component** Create the VTEX Store Component _Product Summary_

File: react/ProductFactory.js [hunk 2]
```diff
@@ -10,6 +10,7 @@ const DEFAULT_PRODUCT = {
   referenceCode: 'ReferenceCode',
```
Comment (brunojdo) [inline react/ProductFactory.js]: url: 'https://www.vtex.com' :trollface:

File: react/ProductSummary.js [hunk 3]
```diff
@@ -131,6 +155,15 @@ ProductSummary.propTypes = {
     /** Product's reference code of the product */
     referenceCode: PropTypes.string,
   }),
+  /** Order form used in the buy button */
+  orderForm: PropTypes.shape({
+    /** The user's cart id */
```
Comment (brunojdo) [inline react/ProductSummary.js]: Remove "The"

File: react/ProductSummary.js [hunk 4]
```diff
@@ -131,6 +155,15 @@ ProductSummary.propTypes = {
...
+    orderFormId: PropTypes.string.isRequired,
+    /** Channel */
+    salesChannel: PropTypes.string.isRequired,
+    /** Which seller is being referenced by the button */
+    seller: PropTypes.string.isRequired,
+  }),
   /** Shows the product list price */
   showListPrice: PropTypes.bool,
   /** If true, shows the pricing labels. If false, only the numbers will be shown */
```
Comment (brunojdo) [inline react/ProductSummary.js]: I think is better somenthing like this: `/** Set pricing labels visibility */`

File: react/ProductSummary.js [hunk 5]
```diff
@@ -154,23 +187,23 @@ ProductSummary.schema = {
   properties: {
     showListPrice: {
       type: 'boolean',
-      title: 'Show product\'s list price',
+      title: "Show product's list price",
     },
     showLabels: {
       type: 'boolean',
-      title: 'Show product\'s prices\' labels (if false, shows only the prices)',
+      title: "Show product's prices' labels (if false, shows only the prices)",
```
Comment (brunojdo) [inline react/ProductSummary.js]: Don't need this comment `(if false, shows only the prices)`

File: react/index.js [hunk 6]
```diff
@@ -3,6 +3,11 @@ import { ExtensionPoint } from 'render'
 
 // FIXME: Delete when we can use true data.
```
Comment (brunojdo) [inline react/index.js]: Is it time? 😄

File: react/ProductSummary.js [hunk 7]
```diff
@@ -154,23 +187,23 @@ ProductSummary.schema = {
...
     showBadge: {
       type: 'boolean',
       title: 'Show the discount badge',
     },
     badgeText: {
       type: 'string',
-      title: 'Badge\'s text',
+      title: "Badge's text",
     },
     hideBuyButton: {
```
Comment (brunojdo) [inline react/ProductSummary.js]: Add a prop to edit text label of the buy button. Something like `editButtonLabel`

File: react/ProductFactory.js (hunk 2 above)
Comment (estacioneto) [reply react/ProductFactory.js]: damn git review line

File: react/index.js (hunk 6 above)
Comment (estacioneto) [reply react/index.js]: If I delete it, we may not see the component working properly... So... Delete this?

File: react/ProductSummary.js [hunk 8]
```diff
@@ -49,37 +52,45 @@ class ProductSummary extends Component {
...
+          <div
+            className="tc pointer"
             onMouseEnter={this.handleMouseEnter}
             onMouseLeave={this.handleMouseLeave}>
-            {!product &&
-            <div>
-              <FormattedMessage id="loading" />
-            </div>
-            }
+            {!product && (
```
Comment (ClaudivanFilho) [inline react/ProductSummary.js]: I think there is no need of this loading here :x

File: react/ProductSummary.js [hunk 9]
```diff
@@ -89,15 +100,28 @@ class ProductSummary extends Component {
...
+                  <div
+                    className={`${
+                      !showButtonOnHover ||
+                      (showButtonOnHover && this.state.isHovering)
+                        ? 'db'
+                        : 'dn'
+                    }`}>
                     {!hideBuyButton && (
-                    // TODO: Use the buy button component
+                      // TODO: Use the buy button component
```
Comment (ClaudivanFilho) [inline react/ProductSummary.js]: remove it

File: react/locales/pt-BR.json [hunk 10]
```diff
@@ -2,5 +2,6 @@
   "pricing.installment-display": "ou até {installments}X de {installmentPrice}",
   "pricing.from": "de:",
   "pricing.to": "por:",
-  "loading": "Carregando…"
+  "loading": "Carregando…",
+  "buy-button": "Comprar"
```
Comment (brunojdo) [inline react/locales/pt-BR.json]: button-label

**Comment by brunojdo** [comment]
Add this Billing Policy on `manifest.json`
```
{
    "billingListeningRoute": "/_v/deactivateMePlease",
    "termsURL": "http://route.to.the/terms",
    "version": "0.1",
    "free": true
}
```

File: react/DiscountBadge.js [hunk 11]
```diff
@@ -16,13 +16,15 @@ class DiscountBadge extends Component {
     const percent = this.calculateDiscountTax(listPrice, sellingPrice)
     return percent ? (
       <div className="vtex-discount-badge relative dib">
-        <div className="f7 dark-gray absolute right-0 pa2-s bg-white">
-          { label === '' && '-' }
-          <FormattedNumber value={percent} style="percent" /> { label }
+        <div className="f7 white absolute right-0 pa2-s bg-red">
```
Comment (leticiawanderley) [inline react/DiscountBadge.js]: Did Pamela agree with this colours?

File: react/ProductSummary.js [hunk 12]
```diff
@@ -27,85 +26,98 @@ class ProductSummary extends Component {
...
+                showListPrice={showListPrice}
+                showLabels={showLabels}
+                showInstallments={showInstallments}
+              />
+            </div>
+          </div>
+          <div className="pv2">
+            <div
+              className={`${
+                !showButtonOnHover ||
```
Comment (leticiawanderley) [inline react/ProductSummary.js]: This expression could be simplified to `(!showButtonOnHover || this.state.hovering)`

File: react/DiscountBadge.js (hunk 11 above)
Comment (brunojdo) [reply react/DiscountBadge.js]: She will check today.

File: react/ProductSummary.js [hunk 13]
```diff
@@ -131,16 +144,27 @@ ProductSummary.propTypes = {
...
+    salesChannel: PropTypes.string.isRequired,
+    /** Which seller is being referenced by the button */
+    seller: PropTypes.string.isRequired,
+  }),
   /** Shows the product list price */
   showListPrice: PropTypes.bool,
-  /** If true, shows the pricing labels. If false, only the numbers will be shown */
+  /** Set pricing labels visibility */
   showLabels: PropTypes.bool,
   /** If true, shows the install information */
```
Comment (brunojdo) [inline react/ProductSummary.js]: Set visibility of installments information

File: react/ProductSummary.js [hunk 14]
```diff
@@ -131,16 +144,27 @@ ProductSummary.propTypes = {
...
   showInstallments: PropTypes.bool,
   /** If true, shows the discount badge */
```
Comment (brunojdo) [inline react/ProductSummary.js]: Set visibility of discount badge

File: react/DiscountBadge.js (hunk 11 above)
Comment (oldufre) [reply react/DiscountBadge.js]: @leticiawanderley, vi agora isso, n sei usar o git :)
Mas sim,  estava discutindo com Bruno e decidimos deixar quase tudo em tons de cinza e só algumas coisas com cores, coisas que devem chamar atenção.
Já que eles já estão usando o styleguide e muitas coisas já possuem as cores, verde, vermelho e azul, vamos usar elas.

Comment (leticiawanderley) [reply react/DiscountBadge.js]: Tá ok :) perguntei porque esse vermelho a gente usa para indicar erro no admin daqui, justamente porque chama atenção.

**Pull request merged**